### PR TITLE
Add GCR staging repo for kubernetes/cloud-provider-azure

### PR DESCRIPTION
**Pull request by nilo19**
Add GCR staging repo for kubernetes/cloud-provider-azure.

### CODE REVIEW

**Comment by feiskyer** [comment]
@justaugustus @craiglpeters could you help to take a look?

File: groups/groups.yaml [hunk 1]
```diff
@@ -810,6 +810,26 @@ groups:
...
+      AllowExternalMembers: "true"
+      WhoCanJoin: "INVITED_CAN_JOIN"
+      WhoCanViewMembership: "ALL_MEMBERS_CAN_VIEW"
+      WhoCanViewGroup: "ALL_MEMBERS_CAN_VIEW"
+      WhoCanDiscoverGroup: "ALL_MEMBERS_CAN_DISCOVER"
+      WhoCanInvite: "NONE_CAN_INVITE"
+      WhoCanAdd: "NONE_CAN_ADD"
+      WhoCanApproveMembers: "NONE_CAN_APPROVE"
+      WhoCanModifyMembers: "NONE"
+      WhoCanModerateMembers: "NONE"
```
Comment (justaugustus) [inline groups/groups.yaml]: Remove these lines. See #447, #446.

File: groups/groups.yaml [hunk 2]
```diff
@@ -810,6 +810,26 @@ groups:
...
+      ReconcileMembers: "true"
```
Comment (justaugustus) [inline groups/groups.yaml]:
```suggestion
      WhoCanPostMessage: "ANYONE_CAN_POST"
      ReconcileMembers: "true"
```

File: infra/gcp/ensure-staging-storage.sh [hunk 3]
```diff
@@ -46,6 +46,7 @@ STAGING_PROJECTS=(
     cluster-api-aws
     cluster-api-azure
     cluster-api-gcp
+    provider-azure
```
Comment (justaugustus) [inline infra/gcp/ensure-staging-storage.sh]: This list needs to be alpha-sorted.

File: groups/groups.yaml [hunk 4]
```diff
@@ -810,6 +810,26 @@ groups:
...
+    members:
+      - t-qini@microsoft.com
+      - Pengfei.Ni@microsoft.com
```
Comment (justaugustus) [inline groups/groups.yaml]:
```suggestion
      - Pengfei.Ni@microsoft.com
      - Stephen@agst.us
```

Comment (justaugustus) [reply groups/groups.yaml]: This list needs to include everyone from https://github.com/kubernetes/cloud-provider-azure/blob/master/OWNERS.

File: k8s.gcr.io/images/k8s-staging-provider-azure/OWNERS [hunk 5]
```diff
@@ -0,0 +1,13 @@
+# See the OWNERS docs at https://go.k8s.io/owners
+# Should always mirror the maintainers/reviewers in https://sigs.k8s.io/cluster-api-provider-azure/OWNERS_ALIASES
```
Comment (justaugustus) [inline k8s.gcr.io/images/k8s-staging-provider-azure/OWNERS]:
```suggestion
# Should always mirror the maintainers/reviewers in https://sigs.k8s.io/cloud-provider-azure/OWNERS
```

Comment (justaugustus) [reply k8s.gcr.io/images/k8s-staging-provider-azure/OWNERS]: The code suggestion here will lead to a dead link, but that will be live once we migrate the repo to `kubernetes-sigs`.

File: k8s.gcr.io/images/k8s-staging-provider-azure/OWNERS [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+
+approvers:
```
Comment (justaugustus) [inline k8s.gcr.io/images/k8s-staging-provider-azure/OWNERS]: This list needs to include everyone from https://github.com/kubernetes/cloud-provider-azure/blob/master/OWNERS.

**Comment by nilo19** [comment]
> @nilo19 -- Thanks for adding this! I've left some comments and code suggestions. :)

Thank you for the suggestions. I have amended the commit.

**Comment by justaugustus** [comment]
Thanks @nilo19!
/lgtm
/approve

**Comment by dims** [comment]
/assign @cblecker 
/unassign

Need to check with @cblecker if we are ready now with the latest updates to the harness/groups.yaml

**Comment by feiskyer** [comment]
kindly ping  @cblecker

File: groups/groups.yaml (hunk 2 above)
Comment (cblecker) [reply groups/groups.yaml]: We don't use these as e-mail DLs, so you can drop the `WhoCanPostMessage` line completely

**Comment by nilo19** [comment]
@cblecker Would you help to review? Thank you.

File: groups/groups.yaml [hunk 7]
```diff
@@ -810,6 +810,23 @@ groups:
       - hi@amycod.es
       - prignanov@vmware.com
 
+  - email-id: k8s-infra-staging-provider-azure@kubernetes.io
+    name: k8s-infra-staging-provider-azure
+    description: |-
+      ACL for staging cloud provider Azure
+    settings:
+      WhoCanPostMessage: "ANYONE_CAN_POST"
```
Comment (justaugustus) [inline groups/groups.yaml]:
```suggestion
```

@nilo19 -- this line still needs to be dropped.
ref: @cblecker's [review](https://github.com/kubernetes/k8s.io/pull/440/files#r341828055)

File: groups/groups.yaml [hunk 8]
```diff
@@ -810,6 +810,23 @@ groups:
...
+      ReconcileMembers: "true"
+    members:
+      - t-qini@microsoft.com
+      - Pengfei.Ni@microsoft.com
+      - Stephen@agst.us
```
Comment (justaugustus) [inline groups/groups.yaml]:
```suggestion
      - stephen.k8s@agst.us
```

I've updated my email address for Kubernetes ACLs.
ref: https://github.com/kubernetes/k8s.io/pull/458

Comment (nilo19) [reply groups/groups.yaml]: Thanks.

**Comment by justaugustus** [comment]
Thanks @nilo19!
@cblecker -- This looks ready to go now.

/lgtm
/approve

**Comment by feiskyer** [comment]
kindly ping @cblecker

**Comment by nilo19** [comment]
Would you help to create the necessary google group? @dims @cblecker

**Comment by nilo19** [comment]
Would you help to create the staging repo?@thockin @justinsb

**Comment by justaugustus** [comment]
> Would you help to create the necessary google group?
> Would you help to create the staging repo?

@nilo19 -- They've already been created:
- https://groups.google.com/a/kubernetes.io/forum/#!forum/k8s-infra-staging-provider-azure
- https://console.cloud.google.com/gcr/images/k8s-staging-provider-azure?project=k8s-staging-provider-azure

You'll be able to view the images once one is actually pushed.

**Pull request merged**